src/Controller/UI/ReportResearchGroupDatasetStatusController.php

```diff
@@ -77,52 +78,152 @@ public function defaultAction(Request $request, EntityHandler $entityHandler, in
+
     /**
      * This method gets data for the report.
      *
-     * @param array|NULL $options Additional parameters needed to run the query.
+     * @param array $options Additional parameters needed to run the query.
      *
      * @return array  Return the data array
      */
-    protected function getData(array $options = null)
+    protected function getData(array $options): array
     {
-        $datasetCountString = 'No datasets';
         $datasets = $options['researchGroup']->getDatasets();
-        $datasetCount = count($datasets);
-        if ($datasetCount > 0) {
-            $datasetCountString = ' [ ' . (string) count($datasets) . ' ]';
+        $reportData = array();
+
+        if (isset($options['version'])) {
+            $reportData = $this->getVersionTwoReport($datasets, $options);
+        } else {
+            $reportData = $this->getVersionOneReport($datasets, $options);
         }
+        return array_merge($this->getDefaultHeaders(), $reportData['additionalHeaders'], $reportData['labels'], $reportData['dataArray']);
+    }
+
+    /**
+     * Create a CSV download filename that contains the truncated research group name and the date/timeto.
+     *
+     * @param string $researchGroupName The name of the Research Group which is the subject of the report.
+     * @param string $researchGroupId   The ID of the Research Group which is the subject of the report.
+     *
+     * @return string
+     */
+    private function createCsvReportFileName(string $researchGroupName, string $researchGroupId)
+    {
+        $nowDateTimeString = date(self::REPORTFILENAMEDATETIMEFORMAT);
+        $researchGroupNameSubstring = substr($researchGroupName, 0, self::MAXRESEARCHGROUPLENGTH);
+        $tempFileName = $researchGroupNameSubstring . '_' . $researchGroupId
+            . '_'
+            . $nowDateTimeString
+            . '.csv';
+        return str_replace(' ', '_', $tempFileName);
+    }
 
+    /**
+     * Get data for version one report.
+     *
+     * @param Collection $datasets  Collection of Datasets.
+     * @param array      $options   Options for report.
+     *
+     * @return array
+     */
+    private function getVersionOneReport(Collection $datasets, array $options): array
+    {
+        $datasetCount = $this->getDatasetCount($datasets);
         //extra headers to be put in the report
         $additionalHeaders = array(
-          array('RESEARCH GROUP',$options['researchGroup']->getName()),
-          array('DATASET COUNT', $datasetCountString),
-          array(parent::BLANK_LINE));
+            array('RESEARCH GROUP',$options['researchGroup']->getName()),
+            array('DATASET COUNT', $datasetCount['string']),
+            array(parent::BLANK_LINE));
 
         //prepare label array
         $labels = array('labels' => array('DATASET UDI',
-          'TITLE',
-          'PRIMARY POINT OF CONTACT',
-          'STATUS',
-          'DATE APPROVED',
-          'DATE REGISTERED'));
+            'TITLE',
```

Comment:
Fields missing: DOI, Restricted Status.

Reply:
This is the version 1 report, which was already there, i just had to refactor it, because we were using almost the same stuff. If you look at Line 292, you can find the labels for the other report which is requested in this ticket

Reply:
I missed that. sry.